### PR TITLE
C++ codes use strictly cmath instead of math.h

### PR DESCRIPTION
**Pull request by ye-luo**
rename ../src/config/stdlib/math.h to ../src/config/stdlib/math.hpp to avoid confusion
implement qmcplusplus::iszero using C++11 functions only. The implementation is near identical to math.h from glibc.
The non standand iszero from glibc is no more used.

### CODE REVIEW

File: src/QMCHamiltonians/SpaceGrid.cpp [hunk 1]
```diff
@@ -15,7 +15,7 @@
 #include <OhmmsData/AttributeSet.h>
 #include <Utilities/string_utils.h>
 #include <cmath>
```
Comment (markdewing) [inline src/QMCHamiltonians/SpaceGrid.cpp]: Duplicate include of cmath

**Comment by prckent** [comment]
Many crashes in AFQMC tests.

**Comment by ye-luo** [comment]
My second commit only removed a duplicated include<cmath>. I can't understand how the first CI passed, the second CI failed.

**Comment by ye-luo** [comment]
Test this please

**Comment by ye-luo** [comment]
AFQMC unit tests run fine on my workstation.

**Comment by ye-luo** [comment]
quirks didn't appear this time. For this PR, I have nothing really special for @PDoakORNL to review.

**Pull request merged**